### ISSUE TITLE
Won't do anything on clear Debian 9.4

### ISSUE THREAD

**Issue by KRtekTM** [AsamK/signal-cli#131]
Hello everybody,

I'm trying to create signal bot using this signal-cli however I'm unable to run it.

I have clean installation of Debian 9.4 with already install default-jre and default-jre-headless package. I also configured the java to have unlimited strength crypto.
Then I downloaded signal-cli 0.6 (and tried to use also 0.5.6), unpacked it following steps in readme file and tried to fire the register command.

It hangs forever in like running state and never return any response (like some error output or something). I'm unable to find any log so I really don't know what is happening there.

I hope that I screwed some configuration or missing some prerequisity.
Please can you help me to reach any additional info about what is happening, so I'll be able to run the software?

Big thanks.

**Comment by Trolldemorted** [comment]
Are you using a server or a desktop installation? Is urandom empty?

**Comment by KRtekTM** [comment]
I'm using Debian desktop with xserver disabled.
echo $(( urandom )) = 0

I have signal-cli stored under ~/signal-cli/bin, tried also another locations, no luck :(

**Comment by KRtekTM** [comment]
Output of jstack reading "hanged" signal-cli while register method is used:

^Croot@vpnserver:/home/ok# jstack 5843
2018-05-23 18:15:00
Full thread dump OpenJDK 64-Bit Server VM (25.171-b11 mixed mode):

"Attach Listener" #9 daemon prio=9 os_prio=0 tid=0x00007f28dc001000 nid=0x1717 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Service Thread" #7 daemon prio=9 os_prio=0 tid=0x00007f29140cc000 nid=0x16ec runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C1 CompilerThread1" #6 daemon prio=9 os_prio=0 tid=0x00007f29140b7800 nid=0x16eb waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #5 daemon prio=9 os_prio=0 tid=0x00007f29140b4800 nid=0x16ea waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Signal Dispatcher" #4 daemon prio=9 os_prio=0 tid=0x00007f29140b3000 nid=0x16e9 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Finalizer" #3 daemon prio=8 os_prio=0 tid=0x00007f2914080800 nid=0x16e8 in Object.wait() [0x00007f29029e7000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x00000000e0ee1258> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
        - locked <0x00000000e0ee1258> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
        at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:212)

"Reference Handler" #2 daemon prio=10 os_prio=0 tid=0x00007f291407c000 nid=0x16e7 in Object.wait() [0x00007f2902ae8000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x00000000e0ee1298> (a java.lang.ref.Reference$Lock)
        at java.lang.Object.wait(Object.java:502)
        at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
        - locked <0x00000000e0ee1298> (a java.lang.ref.Reference$Lock)
        at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

"main" #1 prio=5 os_prio=0 tid=0x00007f291400b000 nid=0x16e3 runnable [0x00007f291ad6c000]
   java.lang.Thread.State: RUNNABLE
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(FileInputStream.java:255)
        at sun.security.provider.NativePRNG$RandomIO.readFully(NativePRNG.java:424)
        at sun.security.provider.NativePRNG$RandomIO.implGenerateSeed(NativePRNG.java:441)
        - locked <0x00000000e0bb1bc8> (a java.lang.Object)
        at sun.security.provider.NativePRNG$RandomIO.access$500(NativePRNG.java:331)
        at sun.security.provider.NativePRNG$Blocking.engineGenerateSeed(NativePRNG.java:274)
        at java.security.SecureRandom.generateSeed(SecureRandom.java:533)
        at org.bouncycastle.jcajce.provider.drbg.DRBG$HybridSecureRandom.<init>(Unknown Source)
        at org.bouncycastle.jcajce.provider.drbg.DRBG.createBaseRandom(Unknown Source)
        at org.bouncycastle.jcajce.provider.drbg.DRBG.access$200(Unknown Source)
        at org.bouncycastle.jcajce.provider.drbg.DRBG$Default.<clinit>(Unknown Source)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.security.Provider$Service.newInstance(Provider.java:1595)
        at sun.security.jca.GetInstance.getInstance(GetInstance.java:236)
        at sun.security.jca.GetInstance.getInstance(GetInstance.java:164)
        at java.security.SecureRandom.getInstance(SecureRandom.java:288)
        at java.security.SecureRandom.getDefaultPRNG(SecureRandom.java:205)
        at java.security.SecureRandom.<init>(SecureRandom.java:162)
        at org.bouncycastle.jcajce.provider.keystore.bc.BcKeyStoreSpi.<init>(Unknown Source)
        at org.bouncycastle.jcajce.provider.keystore.bc.BcKeyStoreSpi$Std.<init>(Unknown Source)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.security.Provider$Service.newInstance(Provider.java:1595)
        at sun.security.jca.GetInstance.getInstance(GetInstance.java:236)
        at sun.security.jca.GetInstance.getInstance(GetInstance.java:164)
        at java.security.Security.getImpl(Security.java:695)
        at java.security.KeyStore.getInstance(KeyStore.java:848)
        at org.whispersystems.signalservice.internal.util.BlacklistingTrustManager.createFor(BlacklistingTrustManager.java:56)
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.createConnectionHolders(PushServiceSocket.java:821)
        at org.whispersystems.signalservice.internal.push.PushServiceSocket.<init>(PushServiceSocket.java:127)
        at org.whispersystems.signalservice.api.SignalServiceAccountManager.<init>(SignalServiceAccountManager.java:102)
        at org.whispersystems.signalservice.api.SignalServiceAccountManager.<init>(SignalServiceAccountManager.java:78)
        at org.asamk.signal.Manager.init(Manager.java:241)
        at org.asamk.signal.Main.handleCommands(Main.java:133)
        at org.asamk.signal.Main.main(Main.java:84)

"VM Thread" os_prio=0 tid=0x00007f2914074000 nid=0x16e6 runnable

"GC task thread#0 (ParallelGC)" os_prio=0 tid=0x00007f2914020000 nid=0x16e4 runnable

"GC task thread#1 (ParallelGC)" os_prio=0 tid=0x00007f2914021800 nid=0x16e5 runnable

"VM Periodic Task Thread" os_prio=0 tid=0x00007f29140ce800 nid=0x16ed waiting on condition

JNI global references: 227

**Comment by KRtekTM** [comment]
IT WORKS!!! Perfect, very big thanks for your help and time.

I used rng-tools so I got some enthropy and now the signal-cli is working well!


It could be useful to add information about missing enthropy issue to the Ubuntu/Debian manual so another user will prevent to fall in the same situation.
